apicast/src/oauth/keycloak.lua

```diff
@@ -229,17 +239,21 @@ function _M:get_token(service, client)
     }
   }
 
-  -- TODO: maybe use the same method the original request uses
   ngx.req.read_body()
-  local req_body = ngx.req.get_post_args()
+  local params = ngx.req.get_post_args()
 
-  ok, err = _M.token_check_params(req_body)
+  local creds = _M.get_client_credentials(params)
+
+  params.client_id = creds.client_id
+  params.client_secret = creds.client_secret
```

Comment:
Because `token_check_params` needs all params + credentials (if they are not in params, they should be taken from headers). Happy to try something different.